config/opal_check_cflags.m4

```diff
@@ -20,7 +20,8 @@ AC_DEFUN([_OPAL_CFLAGS_FAIL_SEARCH],[
 ])
 
 AC_DEFUN([_OPAL_CHECK_SPECIFIC_CFLAGS], [
-AC_MSG_CHECKING(if $CC supports ([$1]))
+    AC_REQUIRE([AC_PROG_GREP])
```

Comment:
I'm not sure I understand this at all...

Reply:
This one kinda sucked.  Here's the output from running `autoreconf` directly (i.e., exactly as it would have been invoked via `autogen.pl`):

```
$ autoreconf -ivf --warnings=all,no-obsolete,no-override -I config |& tee auto.out
autoreconf: export WARNINGS=all
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal -I config --force -I config
configure.ac:387: error: AC_REQUIRE: circular dependency of AC_PROG_GREP
config/opal_check_cflags.m4:13: _OPAL_CFLAGS_FAIL_SEARCH is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2823: _AC_COMPILE_IFELSE is expanded from...
./lib/autoconf/general.m4:2839: AC_COMPILE_IFELSE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
./lib/autoconf/programs.m4:389: AC_PROG_GREP is expanded from...
config/opal_check_cflags.m4:13: _OPAL_CFLAGS_FAIL_SEARCH is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
config/opal_check_cflags.m4:23: _OPAL_CHECK_SPECIFIC_CFLAGS is expanded from...
config/opal_setup_cc.m4:149: OPAL_SETUP_CC is expanded from...
configure.ac:387: the top level
autom4te: error: /cm/shared/apps/autotools/ac271-am1164-lt246/bin/m4 failed with exit status: 1
aclocal: error: /cm/shared/apps/autotools/ac271-am1164-lt246/bin/autom4te failed with exit status: 1
autoreconf: error: aclocal failed with exit status: 1
```

I don't fully grok how the expansion happened that way (i.e., `AC_PROG_GREP` jumped back into `_OPAL_CFLAGS_FAIL_SEARCH`), but by the above output, it apparently does -- perhaps it has something to do with `_OPAL_CFLAGS_FAIL_SEARCH` being invoked twice within `_OPAL_CHECK_SPECIFIC_CFLAGS`...?).

Effectively "moving the `AC_REQUIRE([AC_PROG_GREP])`up" to have the dependency resolved earlier seems to fix the problem.

FWIW, given that `_OPAL_CFLAGS_FAIL_SEARCH` is only invoked within this m4 file, we could also remove the `AC_REQUIRE(...)` from within `_OPAL_CFLAGS_FAIL_SEARCH`, too.

Got any better insight?

![image](https://user-images.githubusercontent.com/617316/134438685-bea28bde-6a9b-4ac1-bcb5-8ae54a5f2445.png)

Reply:
It took longer than I'd like to admit, but I figured it out.  We've underquoted a bunch in the file.  Patch that works:

```
diff --git a/3rd-party/openpmix b/3rd-party/openpmix
--- a/3rd-party/openpmix
+++ b/3rd-party/openpmix
@@ -1 +1 @@
-Subproject commit 5e35d334d7515b23731bb0fbdd504809db339baa
+Subproject commit 5e35d334d7515b23731bb0fbdd504809db339baa-dirty
diff --git a/config/opal_check_cflags.m4 b/config/opal_check_cflags.m4
index 9c454626a1..a7fa7fa866 100644
--- a/config/opal_check_cflags.m4
+++ b/config/opal_check_cflags.m4
@@ -28,11 +28,12 @@ AC_MSG_CHECKING(if $CC supports ([$1]))
                    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [$3])],
                                    [
                                     opal_cv_cc_[$2]=1
-                                    _OPAL_CFLAGS_FAIL_SEARCH("ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown", [$2])
+                                    _OPAL_CFLAGS_FAIL_SEARCH(["ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown"], [$2])
                                    ],
+                                  [
                                     opal_cv_cc_[$2]=1
-                                    _OPAL_CFLAGS_FAIL_SEARCH("ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown\|error", [$2])
-                                 )])
+                                    _OPAL_CFLAGS_FAIL_SEARCH(["ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown\|error"], [$2])
+                                 ])])
             if test "$opal_cv_cc_[$2]" = "0" ; then
                 CFLAGS="$CFLAGS_orig"
                 AC_MSG_RESULT([no])
@@ -59,11 +60,12 @@ AC_MSG_CHECKING(if $CXX supports ([$1]))
                    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [$3])],
                                    [
                                     opal_cv_cxx_[$2]=1
-                                    _OPAL_CXXFLAGS_FAIL_SEARCH("ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown", [$2])
+                                    _OPAL_CXXFLAGS_FAIL_SEARCH(["ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown"], [$2])
                                    ],
+                                  [
                                     opal_cv_cxx_[$2]=1
-                                    _OPAL_CXXFLAGS_FAIL_SEARCH("ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown\|error", [$2])
-                                 )])
+                                    _OPAL_CXXFLAGS_FAIL_SEARCH(["ignored\|not recognized\|not supported\|not compatible\|unrecognized\|unknown\|error"], [$2])
+                                 ])])
             if test "$opal_cv_cxx_[$2]" = "0" ; then
                 CXXFLAGS="$CXXFLAGS_orig"
                 AC_MSG_RESULT([no])
```

Reply:
Sweet!  I'll replace my commit with yours.

Reply:
(BTW, I ignored the submodule update in your patch file; I assume that was unintentionally included -- let me know if there's something you wanted included from openpmix)

Reply:
The openpmix submodule thing was accidental :)